src/glmtools.jl

```diff
@@ -9,28 +9,34 @@ type LogLink      <: Link end
 type ProbitLink   <: Link end
 type SqrtLink     <: Link end
 
+# copied from StatsFuns package to avoid loading the entire package
+xlogy{T<:Real}(x::T, y::T) = x > zero(T) ? x * log(y) : zero(log(x))
+xlogy(x::Real, y::Real) = xlogy(promote(x, y)...)
+
+# x + 1 preserving the type of x
+xp1(x) = x + one(x)
+
 linkfun(::CauchitLink, μ) = tan(pi * (μ - oftype(μ, 1/2)))
 linkinv(::CauchitLink, η) = (oftype(η, 1/2) + atan(η) / pi, false)
-mueta(  ::CauchitLink, η) = one(η) / (pi * (one(η) + abs2(η)))
+mueta(  ::CauchitLink, η) = inv(pi * xp1(abs2(η)))
 
 linkfun(::CloglogLink, μ) = log(-log1p(-μ))
 linkinv(::CloglogLink, η) = η < 0 ? (-expm1(-exp(η)), false) : (exp(-exp(η)), true)
 mueta(  ::CloglogLink, η) = exp(η) * exp(-exp(η))
 
 linkfun(::IdentityLink, μ) = μ
 linkinv(::IdentityLink, η) = (η, false)
-mueta(  ::IdentityLink, η) = 1
+mueta(  ::IdentityLink, η) = one(η)
 
 linkfun(::InverseLink, μ) = inv(μ)
 linkinv(::InverseLink, η) = (inv(η), false)
 mueta(  ::InverseLink, η) = -inv(abs2(η))
 
-linkfun(::LogitLink, μ) = logit(μ)
-linkinv(::LogitLink, η) = (logistic(-abs(η)), η > 0)
+linkfun(::LogitLink, μ) = log(μ / (one(μ) - μ))
+linkinv(::LogitLink, η) = (inv(xp1(exp(-abs(η)))), η > 0)
```

Comment:
This should be `inv(xp1(exp(abs(η))))` and is the reason why the tests fail. However, I think we should just keep using `StatsFuns` for now. It is a dependency of `Distributions` which `GLM` depends on so it is already loaded anyway.